File: backend/src/models/Reservation.js
import mongoose from 'mongoose';

const schema = {
    customer: {
        type: mongoose.Schema.Types.ObjectId,
        required: true,
        ref: 'User'
    },
    date: {
        type: Date,
        required: true
    },
    offering: {
        type: mongoose.Schema.Types.ObjectId,
        required: true,
        ref: 'Offering'
    },
    quantity: {
        type: Number,
        default: 1, 
        required: true
    }
}

export default mongoose.model('Reservation', schema);

File: app/Valapas/app/src/main/java/com/raulbrumar/valapas/Models/Reservation.java
package com.raulbrumar.valapas.Models;

public class Reservation
{
    private String id;
    private User customer;
    private String date;
    private Offering offering;
    private int quantity;

    public Reservation(String id, User customer, String date, Offering offering, int quantity)
    {
        this.id = id;
        this.customer = customer;
        this.date = date;
        this.offering = offering;
        this.quantity = quantity;
    }

    public User getCustomer()
    {
        return customer;
    }

    public String getDate()
    {
        return date;
    }

    public Offering getOffering()
    {
        return offering;
    }

    public int getQuantity()
    {
        return quantity;
    }
}
